.github/workflows/cleanup-caches.yml

```diff
@@ -0,0 +1,43 @@
+##
+# Clean up cache keys that are no longer needed.
+##
+name: Cache Garbage Collection
+
+on:
+  pull_request:
```

Comment:
is this here for testing? later on maybe we run this on merge into trunk? or only when closing a PR?

Reply:
Ya, sorry. I forgot to re-add the `types` filter for only running on `closed` after testing. All set now!